internal/errgrouputil/builder.go

```diff
@@ -20,7 +20,7 @@ func Build[T any](
 	builders ...BuilderFunc[T],
 ) ([]*T, []error) {
 	eg, ctx := errgroup.WithContext(ctx)
-	eg.SetLimit(runtime.NumCPU()/2 + 1)
+	eg.SetLimit(runtime.GOMAXPROCS(0)/2 + 1)
```

Comment:
Should this consider both GOMAXPROCS and NumCPU, and base the limit on whichever is lower?

Reply:
I'm not sure what the purpose of this was. It sounds like it was put in place to avoid CPU starvation, but Go has had a pre-emptive scheduler for a while now, so starvation shouldn't be an issue.

@wasaga what do you think it should do?